x/gov/ante/gov_filter.go

```diff
@@ -0,0 +1,91 @@
+package ante
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	govkeeper "github.com/cosmos/cosmos-sdk/x/gov/keeper"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+var MiniumInitialDepositRate = sdk.NewDecWithPrec(10, 2)
+
+type GovPreventSpamDecorator struct {
+	govKeeper *govkeeper.Keeper
+	cdc       codec.BinaryCodec
+}
+
+func NewGovPreventSpamDecorator(cdc codec.BinaryCodec, govKeeper *govkeeper.Keeper) GovPreventSpamDecorator {
+	return GovPreventSpamDecorator{
+		govKeeper: govKeeper,
+		cdc:       cdc,
+	}
+}
+
+func (gpsd GovPreventSpamDecorator) AnteHandle(
+	ctx sdk.Context, tx sdk.Tx,
+	simulate bool, next sdk.AnteHandler,
+) (newCtx sdk.Context, err error) {
+	msgs := tx.GetMsgs()
+	if err = gpsd.checkSpamSubmitProposalMsg(ctx, msgs); err != nil {
+		return ctx, err
+	}
+
+	return next(ctx, tx, simulate)
+}
+
+func (gpsd GovPreventSpamDecorator) checkSpamSubmitProposalMsg(ctx sdk.Context, msgs []sdk.Msg) error {
+	validMsg := func(m sdk.Msg) error {
+		switch msg := m.(type) { //nolint:gocritic
```

Comment:
Doesn't seem to need to be a switch statement here, simple if statement to check if the `msg.type` is `MsgSubmitProposal`